src/services/notifications/notifications.ts

```diff
@@ -0,0 +1,146 @@
+import type { HttpInstance } from '../../types';
+import { AffectedRecords, PagedResult, ObjectId } from '../types';
+import { RQLString, rqlBuilder } from '../../rql';
+import type {
+  Notification,
+  CreateNotificationRequest,
+  NotifTypeDef,
+} from './types';
+
+export default (client, httpAuth: HttpInstance) => ({
+  /**
+   * Retrieve a list of your own notifications
+   * Permission | Scope | Effect
+   * - | - | -
+   * none | | Everyone can use this endpoint
+   *
+   * @param rql Add filters to the requested list.
+   * @returns PagedResult<Notification>
+   */
+  async find(options?: {
+    rql?: RQLString;
+  }): Promise<PagedResult<Notification>> {
+    return (await client.get(httpAuth, `/${options?.rql || ''}`)).data;
+  },
+
+  /**
+   * Find By Id
+   * @param id the Id to search for
+   * @param rql an optional rql string
+   * @returns the first element found
+   */
+  async findById(
+    id: ObjectId,
+    options?: { rql?: RQLString }
+  ): Promise<Notification> {
+    const rqlWithId = rqlBuilder(options?.rql).eq('id', id).build();
+    const res = await this.find({ rql: rqlWithId });
+    return res.data[0];
+  },
+
+  /**
+   * Find First
+   * @param rql an optional rql string
+   * @returns the first element found
+   */
+  async findFirst(options?: { rql?: RQLString }): Promise<Notification> {
+    const res = await this.find(options);
+    return res.data[0];
+  },
+
+  /**
+   * Create a notification
+   * Permission | Scope | Effect
+   * - | - | -
+   * none | | Create a notification for yourself
+   * `CREATE_NOTIFICATIONS` | `global` | Create a notification for another person
+   *
+   * @param requestBody CreateNotificationRequest
+   * @returns Notification
+   */
+  async create(requestBody: CreateNotificationRequest): Promise<Notification> {
+    return (await client.post(httpAuth, '/', requestBody)).data;
+  },
+
+  /**
+   * Retrieve a list of notifications
+   * Permission | Scope | Effect
+   * - | - | -
+   * none | | View your own notifications
+   * `VIEW_NOTIFICATIONS` | `global` | View all notifications
+   *
+   * @param rql Add filters to the requested list.
+   * @returns PagedResult<Notification>
+   */
+  async findAll(options?: {
+    rql?: RQLString;
+  }): Promise<PagedResult<Notification>> {
+    return (await client.get(httpAuth, `/notifications${options?.rql || ''}`))
+      .data;
+  },
+
+  /**
+   * Find By Id
+   * @param id the Id to search for
+   * @param rql an optional rql string
+   * @returns the first element found
+   */
+  async findAllById(
```

Comment:
Might sound confusing? Maybe we can discuss naming during call? In other services there's just one 1 endpoint for your own / others

Reply:
cool, has been removed now following the discussion with Kobe